backend/baseclass.pm

```diff
@@ -1219,21 +1228,72 @@ sub start_ssh_serial {
 
 sub check_ssh_serial {
     my ($self, $fh) = @_;
+    my $read_size = 1024;
+    my $buffer;
 
     if ($self->{serial} && $self->{serial}->sock == $fh) {
-        my $chan = $self->{serial_chan};
-        my $line = <$chan>;
-        if (defined $line) {
-            print $line;
+        if (defined($self->{serial_chan}->read($buffer, $read_size))) {
```

Comment:
Not sure if this could cause problems in other places, my tests didn't show any. 
For me it is much more usual to read the output in blocks, then line by line. We also avoid a warning with this change:
```
Calling Net::SSH2::Channel::readline in non-blocking mode is usually a programming error
```